fw/Core/Hitcon/Logic/Display/display.cc

```diff
@@ -53,44 +53,52 @@ void get_scroll_frame(uint8_t *buf, int frame) {
    *                                                            +---+
    */
 
-  int total_width = DISPLAY_WIDTH + display_scroll_info.cols + 1;
+  int total_width = DISPLAY_WIDTH + display_scroll_info.n_col + 1;
   int period = total_width * display_scroll_info.speed;
   int x_at_frame0 = -DISPLAY_WIDTH;
   int current_x = x_at_frame0 + (frame - display_scroll_info.first_frame) %
                                     period / display_scroll_info.speed;
 
   for (int y = 0; y < DISPLAY_HEIGHT; y++) {
     for (int x = 0; x < DISPLAY_WIDTH; x++) {
-      buf[y * DISPLAY_WIDTH + x] =
-          (0 <= current_x + x && current_x + x < display_scroll_info.cols)
-              ? display_scroll_info
-                    .buf[y * display_scroll_info.cols + current_x + x]
-              : 0;
+      int inside_window =
+          (0 <= current_x + x && current_x + x < display_scroll_info.n_col);
+      display_buf_assign(
+          buf[x], y,
+          inside_window
+              ? display_buf_get(display_scroll_info.buf[current_x + x], y)
+              : 0);
```

Comment:
don't know why need a extra int and a ternary operator here.

Reply:
For readability. You can suggest a better option.